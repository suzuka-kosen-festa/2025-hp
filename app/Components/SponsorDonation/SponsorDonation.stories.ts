import type { Meta, StoryObj } from "@storybook/react";
import SponsorDonation from "./SponsorDonation";

const meta: Meta<typeof SponsorDonation> = {
  title: "Components/SponsorDonation",
  component: SponsorDonation,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
} satisfies Meta<typeof SponsorDonation>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
