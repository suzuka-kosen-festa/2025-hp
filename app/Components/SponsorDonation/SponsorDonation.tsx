import type { FC } from "react";
import { Box, Button, Link, Typography } from "@mui/material";
import { styled } from "@mui/material/styles";
import { ja } from "@/locales/ja";

const DonationButton = styled(Button)(({ theme }) => ({
  "background": "linear-gradient(45deg, #FF6B6B 30%, #FF8E53 90%)",
  "color": "white",
  "fontWeight": "bold",
  "padding": theme.spacing(1.5, 4),
  "borderRadius": "25px",
  "fontSize": "1.1rem",
  "textTransform": "none",
  "boxShadow": "0 4px 15px rgba(255, 107, 107, 0.3)",
  "&:hover": {
    background: "linear-gradient(45deg, #FF5252 30%, #FF7043 90%)",
    boxShadow: "0 6px 20px rgba(255, 107, 107, 0.4)",
    transform: "translateY(-2px)",
  },
  "transition": "all 0.3s ease",
}));

const SponsorDonation: FC = () => {
  return (
    <Box sx={{ p: { xs: 2, sm: 3, md: 4 }, textAlign: "center" }}>
      <Typography
        variant="h4"
        component="h2"
        sx={{
          color: "white",
          fontWeight: "bold",
          mb: 3,
          fontSize: { xs: "1.5rem", sm: "2rem", md: "2.5rem" },
          textShadow: "0 2px 4px rgba(0, 0, 0, 0.3)",
        }}
      >
        {ja.sponsorDonation.title}
      </Typography>

      <Typography
        variant="body1"
        sx={{
          color: "white",
          mb: 3,
          fontSize: { xs: "1rem", sm: "1.1rem" },
          lineHeight: 1.8,
          textShadow: "0 1px 2px rgba(0, 0, 0, 0.3)",
        }}
      >
        {ja.sponsorDonation.description}
        <br />
        {ja.sponsorDonation.benefit}
      </Typography>

      <Typography
        variant="body2"
        sx={{
          color: "rgba(255, 255, 255, 0.95)",
          mb: 4,
          fontSize: { xs: "0.9rem", sm: "1rem" },
        }}
      >
        {ja.sponsorDonation.note}
      </Typography>

      <Link
        href="https://forms.gle/4TyeBZ2iVsNY9oKb8"
        target="_blank"
        rel="noopener noreferrer"
        sx={{ textDecoration: "none" }}
      >
        <DonationButton variant="contained" size="large">
          {ja.sponsorDonation.buttonText}
        </DonationButton>
      </Link>

      <Typography
        variant="caption"
        sx={{
          color: "rgba(255, 255, 255, 0.7)",
          display: "block",
          mt: 2,
          fontSize: { xs: "0.8rem", sm: "0.9rem" },
        }}
      >
        {ja.sponsorDonation.linkDescription}
      </Typography>
    </Box>
  );
};

export default SponsorDonation;
