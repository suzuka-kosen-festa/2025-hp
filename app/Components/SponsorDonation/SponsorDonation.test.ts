// @vitest-environment jsdom

import { composeStories } from "@storybook/react";
import { expect, it } from "vitest";
import * as stories from "./SponsorDonation.stories";

const { Default } = composeStories(stories);

it("sponsorDonation snapshot", async () => {
  await Default.run();
  expect(document.body.firstChild).toMatchSnapshot();
});
