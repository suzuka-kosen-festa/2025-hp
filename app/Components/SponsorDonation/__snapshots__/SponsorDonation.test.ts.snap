// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sponsorDonation snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-19by9mf"
  >
    <h2
      class="MuiTypography-root MuiTypography-h4 css-puoz35-MuiTypography-root"
    >
      個人協賛のお願い
    </h2>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1cgly3f-MuiTypography-root"
    >
      第60回鈴鹿高専祭では、個人の皆様からの協賛金を募集しております。
      <br />
      ご支援いただいた方には御礼として公式サイトおよび会場内へ名前を掲載させていただきます。
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 css-meflsb-MuiTypography-root"
    >
      ※企業・団体様は、お手数ですが下記リンク先に記載のメールアドレスまでご連絡ください。
    </p>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1marwbi-MuiTypography-root-MuiLink-root"
      href="https://forms.gle/4TyeBZ2iVsNY9oKb8"
      rel="noopener noreferrer"
      target="_blank"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-1ocd3b6-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        協賛金フォームへ
      </button>
    </a>
    <span
      class="MuiTypography-root MuiTypography-caption css-1ptwzt3-MuiTypography-root"
    >
      詳しくは上記リンクよりご確認ください
    </span>
  </div>
</div>
`;
